openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -57,10 +58,22 @@ def process(self, instance):
         else:
             members = instance[:]
 
+        loaded_containers = None
+        if {f.lower() for f in self.add_for_families}.intersection(
```

Comment:
Why should this be non case sensitive in the check even though families in Pyblish plugins are case sensitive, right?

Also, are we missing `self.add_for_families` attribute being explicitly set in the code? (Or am I just missing it in the GitHub view somehow?)

Instead of adding these in the extractor - shouldn't these originally be "collected" along instead?

Reply:
`add_for_families` is [added there directly from Settings](https://github.com/pypeclub/OpenPype/blob/1ccc2ad45822f3b9c5b2db29e3000ba158bb5c2b/openpype/lib/plugin_tools.py#L267-L354), no need to manually specify.

With case sensitivity - I am not fan of case sensitive family names as I belive it just adds to chaos - there is no reason why `layout` should be different from `Layout` even more when you allow user to set family name by typing it somewhere in the Settings.

Reply:
I'd avoid the case-insensitivity here. Pyblish plug-in families are case-sensitive. This logic is tied very close to those exact `families`. There are more OpenPype settings exposed that related to Pyblish families and those are also case sensitive.

It's less code/logic to maintain, less cases to test and also more explicit. We're only making things more complicated by allowing the other cases. Plus I'd personally rather not see code with `LaYouT` and have it considered valid too.

To me plug-in families are constants. If we want to avoid human error in the settings then it should be an enum with a list to specify from. User might just as well mistype with a different character as recently was found with aftereffects being mistyped.

> add_for_families is [added there directly from Settings](https://github.com/pypeclub/OpenPype/blob/1ccc2ad45822f3b9c5b2db29e3000ba158bb5c2b/openpype/lib/plugin_tools.py#L267-L354), no need to manually specify.

A code linter will likely not detect that? And most plug-ins that I've seen that specify custom settings also define the attribute on the class for readability. We should find a consistency between those two - I'd prefer the explicit method for readability.

I'd even go as far as not allowing Settings to set attributes on the plug-ins that do not exist yet to enforce this explicit code style.

Reply:
@mkolar Any preference on this from your end? ☝️

Reply:
@mkolar I saw you approved this PR - could you check in regarding this question above?

Reply:
and yes, after giving it more thought than it deserves :) I agree with your point.  But I believe with this we really need to implement some central way of managing families - we are currently unable to present (for example in Settings) all families we have, so in many places you just need to write it down manually and that is source of errors.